File: python/keyword-volatility.py
#
# GitHub: https://github.com/SERPWoo/API
#
# This code requests a Keyword Volatility and outputs the Epoch Timestamp, Volatility %
#
# This output is text format
#
# Last updated - May 5, 2019 @ 20:40 EST (@MercenaryCarter https://github.com/MercenaryCarter and https://twitter.com/MercenaryCarter)
#
# Run Command: python keyword-volatility.py


import requests, json

API_key = 'API_KEY_HERE'
Project_ID = 0 #input your Project ID
Keyword_ID = 0 #input your Keyword ID

url = "https://api.serpwoo.com/v1/volatility/" + str(Project_ID) + "/" + str(Keyword_ID) + "/"
payload = {'key': API_key}

r = requests.get(url, params=payload)
j = r.json()

print('\n--\n')

print('%-20s %-20s' % ("Timestamp",  "Volatility %"))
print('%-20s %-20s' % ("----------", "------------"))

if j['success'] == 1:

        #print('%-15s' % (project_id))
        for timestamp in sorted(j[str(Project_ID)][str(Keyword_ID)], reverse=True):

            print('%-20s %-20s' % (timestamp, j[str(Project_ID)][str(Keyword_ID)][timestamp]['volatility']))

else:

    print('An Error Occurred: ' + j['error'])


print('\n--\n')

File: ruby/keyword-volatility.rb
#!/usr/bin/ruby
#
# GitHub: https://github.com/SERPWoo/API
#
# This code requests a Keyword Volatility and outputs the Epoch Timestamp, Volatility %
#
# This output is text format
#
# Last updated - May, 6th @ 8:15 EST (@MercenaryCarter https://github.com/MercenaryCarter and https://twitter.com/MercenaryCarter)
#
# Run Command: ruby keyword-volatility.rb
#

require 'net/http'
require 'uri'
require 'json'

  # Get your API Key here: https://www.serpwoo.com/q/api/ (should be logged in)
  API_key	= "API_KEY_HERE"
  Project_ID	= 0 #input your Project ID
  Keyword_ID	= 0 #input your Keyword ID

    def open(url)
      Net::HTTP.get(URI.parse(url))
    end

  page_content = open("https://api.serpwoo.com/v1/volatility/#{Project_ID}/#{Keyword_ID}/?key=#{API_key}")
    
  parsed_projects = JSON.parse(page_content)["#{Project_ID}"]
  
  puts "\n--"
  
    printf "%-15s %-20s\n", "Timestamp",  "Volatility %"
    printf "%-15s %-20s\n", "----------", "------------"
  
    parsed_projects.sort.each do |the_keyword, keyword_obj|

      keyword_obj.sort.reverse.each do |the_timestamp, timestamp_obj|

        timestamp_obj.sort.each do |the_volatility, volatility_value|

          if the_volatility == 'volatility'

              printf "%-20s %-20s\n", the_timestamp, volatility_value

            end

        end

      end

    end

  puts "--\n\n"